working/2364 - primary constructors/feature-specification.md

```diff
@@ -202,21 +202,29 @@ enum E {
 class const Point(int x, int y);
 
 enum E(String s) { one('a'), two('b') }
+```
 
-extension type I.name(int x); // Must use a primary constructor.
+Finally, an extension type declaration is specified to use a
+primary constructor (in that case there is no other choice,
+it is in the grammar rules):
+
+```dart
+// Using a primary constructor.
 
+extension type I.name(int x); // Must use a primary constructor.
 ```
 
-This mechanism follows an existing pattern, where `const` modifiers can be
-omitted in the case where the immediately syntactic context implies that
-this modifier _must_ be present. For example, `const [const C()]` can be
-written as `const [C()]`. In the examples above, the parameter-and-variable
-declarations `final int x` and `final int y` are written as `int x` and
-`int y`, and this is allowed because it would be a compile-time error to
-omit `final` in an `extension type`, and in a class with a constant
-constructor. In other words, when we see an `extension type` declaration, or we
-see `const` on the name of a class, we know that `final` is implied on all
-instance variables.
+The primary constructor feature follows an existing pattern, where `const`
+modifiers can be omitted in the case where the immediately syntactic
+context implies that this modifier _must_ be present. 
+
+For example, `const [const C()]` can be written as `const [C()]`. In the
+examples above, the parameter-and-variable declarations `final int x` and
+`final int y` are written as `int x` and `int y`, and this is allowed
+because it would be a compile-time error to omit `final` in an `extension
+type`, and in a class with a constant constructor. In other words, when we
+see an `extension type` declaration, or we see `const` on the name of a
+class, we know that `final` is implied on all instance variables.
```

Comment:
This paragraph starts with "For example", so it sounds descriptive.
However, the content feels prescriptive. 

It's the first notion that `final` can be omitted, and it doesn't follow as "for example" from the previous paragraph which only mentions `const`.

Reply:
I removed the paragraph: There is a certain similarity between the implicit provision of `final` on these implicitly induced variable declarations and the implicit provision of the modifier `const` in nested levels of a constant expression, but they are also quite different. So it's gone.